docs/assignment2.md

```diff
@@ -2,94 +2,359 @@
+The player can use the *command* *Inspect Scene* to receive more information about the interactable objects inside the *scene* they are in, in the *studio scene* that is the two *doors* and the items. \
+The *player* object is connected to the *stats* object which hold the amount of *health points*, which are 50 out of the *maximum health points*: 100.  \
+Using *player commands* will let the player check out their *health points* and what is inside their *inventory*. At the start of the game the player has *a phone* and five *cigarettes* inside the *inventory*.
+
+
+
+Word count: 408
+
+
+
+## State machine diagrams
+Author(s): Claudia Grigoras
+
+<b> Game Class - State Machine Diagram </b>
+![State Machine Diagram - Game](images/SMDGame.jpeg)
+
+This diagram represents the finite number of states for the **game** class. This is the class that initializes the game and that ensures that the commands are being read, are valid and then run. 
+
+The first state after the initial state within the state machine diagram is the "Initialize Game". Within this state, as it can be seen within the code as well, everything is initialized and thus creating objects of all other classes to be called later on within the other states. The main effect (activity) that will be transitioned through this state is the listener, as the listener is used within the next state "Start Game". 
+
+"Start Game" is a composite state, whereas there are a few internal activities. First of all, at the entry the state gets the listener in order to operate other activities. Then in order to actually start the game, the listener is being run (actively listening to changes inputted within the terminal), and the current scene id is set according to the first scene id of the game. Then at exiting this state, a command string has been read. 
+
+This string is transitioned to the next state which is "Read Command". This, same as the previous state, is a composite state. At entry, the state gets the string command, and as an internal activity, checks this command to see whether it is valid or not. If the command is not valid, then the next state is the flow final. If it is valid, at exit, the command is evaluated as being valid. This command is then transitioned to the "Handle Command" state, again a composite state. 
+
+The state "Handle Command" gets the valid command at entry, and then runs the scene for that command (for which the next state machine diagram is being provided below). Upon exiting this state, the scene will change to a new scene and new commands need to be read. Therefore, the state transitions back to "Read Command" if there is an event e = listener where the guard (condition) is g = new command, so that e[g]/string command. These states will loop between themselves repeatedly, unless it reads an invalid command. 
+
```

Comment:
-States should be nouns.
-This is an activity diagram for all intents and purposes.